src/abstractdataframe/reshape.jl

```diff
@@ -355,35 +368,87 @@ julia> unstack(df, :variable, :value, fill=0)
 ─────┼──────────────────────
    1 │ 1           1      2
    2 │ 2           3      0
+
+julia> df = DataFrame(cols=["a", "a", "b"], values=[1, 2, 4])
+3×2 DataFrame
+ Row │ cols    values
+     │ String  Int64
+─────┼────────────────
+   1 │ a            1
+   2 │ a            2
+   3 │ b            4
+
+julia> unstack(df, :cols, :values, operation=copy)
+1×2 DataFrame
+ Row │ a        b
+     │ Array…?  Array…?
+─────┼──────────────────
+   1 │ [1, 2]   [4]
+
+julia> unstack(df, :cols, :values, operation=sum)
+1×2 DataFrame
+ Row │ a       b
+     │ Int64?  Int64?
+─────┼────────────────
+   1 │      3       4
 ```
-Note that there are some differences between the widened results above.
 """
 function unstack(df::AbstractDataFrame, rowkeys, colkey::ColumnIndex,
                  value::ColumnIndex; renamecols::Function=identity,
-                 allowmissing::Bool=false, allowduplicates::Bool=false, fill=missing)
-    rowkey_ints = vcat(index(df)[rowkeys])
+                 allowmissing::Bool=false, allowduplicates::Bool=false,
+                 operation=nothing, fill=missing)
+    if !isnothing(operation)
```

Comment:
no - because we pass a `SubVector` to the function. we could do `operation = last` but it would be slower (as doing the transformation inside the `if` is faster than doing `allowduplicates=true` as I have commented above). Therefore I think using `nothing` as a default is better.